### ISSUE TITLE
Function detection ignores endbr64

### ISSUE THREAD

**Issue by LMMilewski** [Vector35/binaryninja-api#2086]
Binary Ninja Version: 2.2.2487 Personal, e0eb0a77
Platform: Ubuntu 20.04.1 LTS

Consider the following C++ program:
```
void foo(){}
void (*f)() = foo;
int main() {
  f();
  return 0;
}
```
Built with
```
$ g++ --version
g++ (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ g++ repro.cc -o repro
```
And stripped
```
$ strip -s repro
```

Bytes for function foo:
```
0xf3, 0x0f, 0x1e, 0xfa, 0x55, 0x48, 0x89, 0xe5, 0x90, 0x5d, 0xc3
```

Got: Binary Ninja detected the function foo as (note that the first 4 bytes are ignored):
```
0000112d  55                 push    rbp {__saved_rbp}
0000112e  4889e5             mov     rbp, rsp {__saved_rbp}
00001131  90                 nop     
00001132  5d                 pop     rbp {__saved_rbp}
00001133  c3                 retn     {__return_addr}
```
Want: But the correct version would be:
```
00001129  f30f1efa           endbr64 
0000112d  55                 push    rbp {__saved_rbp}
0000112e  4889e5             mov     rbp, rsp {__saved_rbp}
00001131  90                 nop     
00001132  5d                 pop     rbp {__saved_rbp}
00001133  c3                 retn     {__return_addr}
```

**Comment by psifertex** [comment]
I've marked this as a bug, but it's partially to be expected. Binary Ninja does not have hard-coded function start patterns built in. Rather, it uses recursive decent, exports, and other function starts to identify very high probability functions and then once it has identified enough of those, it will use those functions to try to identify other functions in the binary. 

Given that, this behavior isn't particularly surprising for a very small sample file. For a much lager binary with many valid function starts beginning with enbr64, I would expect this behavior would not occur. Did you first observe this bug in a lager binary or was it only observable in very small files?

**Comment by bpotchik** [comment]
Fixed in `2.2.2512-dev`

Thanks for the nice repro. I just needed to reprioritize 1 line of code for the fuzzy prologue detection to pick this up.

**Comment by LMMilewski** [comment]
Thank you for a very quick fix!
